### ISSUE TITLE
Does this work?

### ISSUE THREAD

**Issue by darrell24015** [ubidots/ubidots-arduino-wifly#1]
Hi - I've spent several hours with this code and not getting anything to go to Ubidots.  I'm not collecting any analog data, but rather have a DHT11 for temperature and humidity.  I have my sketch collecting the data from the DHT11 properly and my goal was to use the humidity value.  I have my auth token and the variable ID number for the humidity.

Could you document some of the steps inside the function better (what ID number belongs where) and the call to the save_value() function and debugging code is not explained.

Thanks!

**Comment by AgustinPelaez** [comment]
Hi Darrell,

Here's a more detailed document on the steps we followed:

http://ubidots.com/docs/devices/wifly.html

Please notice that we used the shield from Seeedstudio. Are you also using this shield? Also, which Arduino IDE are you using?

Let us know how it goes; we'll grab our unit and run some tests ourselves to see how we can improve the document.

Thanks,  Agustin

Let us know

**Comment by darrell24015** [comment]
Thanks for the reply!  I'm using the XBee Shield and Wifi Bee v2.0 (WiFly RN171) all from SeeedStudio. I think those are essentially the same components as the Wifly shield. I have the Arduino IDE 1.6.3.

Just to test, I used the sample sketch "wifly.ino" and put in a static int value for the measurement (I don't have anything connected to analog input).  I used my Variable ID and the short Auth ID in the connection string.   I get back "400 Bad Request" when I do the debug.

I'll look at that URL and keep on trying.  Thanks again!

**Comment by darrell24015** [comment]
Also, I'm using the WiFly Library from SeeedStudio, not the Library from Ubidots.  But, it looked like it was the same Library inside the ZIP file (maybe different dates of the files).

**Comment by AgustinPelaez** [comment]
Hi Darrell!

How did your tests go? Are you using this example --> https://github.com/Seeed-Studio/WiFi_Shield/tree/master/Examples/wifly_http ? If so, feel free to send us your code to support@ubidots.com and we'll have a look.

- Agustín

**Comment by darrell24015** [comment]
Hi Agustin,

No, actually, I have been using the example sketch from the Ubidots
repository located here:

https://github.com/ubidots/ubidots-arduino-wifly

Do I need to use the Seeed example to connect to Ubitdots?


Here's is the code that I have been using, just to see if I can post a
static value to Ubidots:

https://github.com/darrell24015/Uno/tree/master/wifly

I get the 400 Bad Request error.  Do I need to use the short auth token or
the long token?

I have another sketch where I am capturing the Humidity and Temperature
from a DHT11 sensor and trying to post to Ubidots:

https://github.com/darrell24015/Uno/tree/master/DHT_Sensor/DHTtester

This also fails.

Thanks for all your help with this! I may have a good project to post if we
get this working!

DL

_________________________

Darrell Little
about.me/darrell_little
  [image: Darrell Little on about.me]
    <http://about.me/darrell_little>

**Comment by AgustinPelaez** [comment]
Hi Darrell,

The line were you setup the POST request (snprintf(buf, sizeof(buf), ....) can be problematic because a slight change in a character could cause the "bad request" response. Try adding more buffer to it, something like:

 char buf[300]; // instead of 200

A better idea would be to use the HTTP client already built into the SeeedStudio library (https://github.com/Seeed-Studio/WiFi_Shield/tree/master/Examples/wifly_http). When we developed the library we were not aware that there was an HTTP library already, so here's an example of how it would look pointing to Ubidots: https://gist.github.com/AgustinPelaez/6b4fbb2b61c9e64a476d)

It's a pitty we don't have the module in our lab right now so we could test (it's installed on site) but I hope this helps.

--AP

**Comment by darrell24015** [comment]
Hi AP,
I changed the buffer to 300 as you suggested.  I no longer get the bad request error, but nothing shows up on my Ubidots data source.  

I also tried the HTTP client (changing it to my own device ID and token) and initially got this error "Path buffer too small. Failed to parse URL"

I went to the source code HTTPClient.h and changed these:
#define HTTP_MAX_PATH_LEN                   264
#define HTTP_MAX_BUF_LEN                    1000

Don't get the error, but the same - nothing shows up in Ubidots.   I'm going to try putting in more code to give more feedback on the status of the HTTP request and see if there's more to learn.

If you get a chance to test this in your lab, please share what you found.

Thanks!

**Comment by darrell24015** [comment]
I worked several hours on this over the weekend. I added some code to handle the error response and even with the HTTP client in your example, I'm getting back the 400 Bad Request error. I read through the entire Ubidots API and it's fails to show an example of a complete URL string to send the sensor data. From what is shown in the API, I was able to post data using Curl on a Linux machine, but of course, I need this to work with an Arduino/WiFly shield instead.

Looking at the HTTPClient source code, I see the post method is overloaded - there is post(url, data, timeout) and post(url, headers, data, timeout).  Perhaps we're missing something in the post method being used?

I'm really looking forward to making this work, it will be exciting when it does happen!
